flutter/shell/platform/tizen/channels/platform_channel.cc

```diff
@@ -95,58 +99,68 @@ void PlatformChannel::HandleMethodCall(
     std::unique_ptr<MethodResult<rapidjson::Document>> result) {
   const std::string& method = method_call.method_name();
   const rapidjson::Document* arguments = method_call.arguments();
+  result_ = std::move(result);
 
   if (method == kSystemNavigatorPopMethod) {
     SystemNavigatorPop();
-    result->Success();
+    result_->Success();
   } else if (method == kPlaySoundMethod) {
     PlaySystemSound(arguments[0].GetString());
-    result->Success();
+    result_->Success();
   } else if (method == kHapticFeedbackVibrateMethod) {
     std::string type;
     if (arguments->IsString()) {
       type = arguments[0].GetString();
     }
     HapticFeedbackVibrate(type);
-    result->Success();
+    result_->Success();
   } else if (method == kGetClipboardDataMethod) {
     // https://api.flutter.dev/flutter/services/Clipboard/kTextPlain-constant.html
     // The API only supports the plain text format.
     if (strcmp(arguments[0].GetString(), kTextPlainFormat) != 0) {
-      result->Error(kUnknownClipboardFormatError,
-                    "Clipboard API only supports text.");
+      result_->Error(kUnknownClipboardFormatError,
+                     "Clipboard API only supports text.");
       return;
     }
-    GetClipboardData([result = result.release()](const std::string& data) {
-      rapidjson::Document document;
-      document.SetObject();
-      rapidjson::Document::AllocatorType& allocator = document.GetAllocator();
-      document.AddMember(rapidjson::Value(kTextKey, allocator),
-                         rapidjson::Value(data, allocator), allocator);
-      result->Success(document);
-      delete result;
-    });
+    if (!GetClipboardData([this](std::optional<std::string> data) {
+          if (!data.has_value()) {
+            result_->Error(kUnknownClipboardError,
+                           "Clipboard.GetData() failed.");
```

Comment:
The error message is not informative (also not `GetData` but `getData`).
```suggestion
                           "Internal error.");
```

Reply:
done